vcpkg.json

```diff
@@ -12,11 +12,5 @@
     {
       "name": "cmocka"
     }
-  ],
-  "overrides": [
-    {
-      "name": "zlib",
-      "version": "1.2.12#2"
-    }
```

Comment:
Reverting https://github.com/Azure/azure-sdk-for-c/pull/2377

With the latest baseline (if it builds successfully), the overrides shouldn't be necessary.